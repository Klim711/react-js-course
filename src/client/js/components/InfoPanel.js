import React, {Component} from 'react';
import {Link} from 'react-router-dom';

const InfoPanel = function({film}) {
  return (
    <header className="row page-header">
      <div className="title-block">
        <h1 className="title">netflixroulette</h1>

        <Link to="/">
          <button className="btn">
            Search
          </button>
        </Link>
      </div>

      <div className="info-panel">
        <div className="info-panel__cover">
          <img src={film.poster_path}/>
        </div>

        <div className="info-panel__text">
          <h2>{film.title} <span>{film.vote_average}</span></h2>

          <div className="additional-info">
            <span className="year">{film.release_date}</span>
            <span className="time">{film.runtime} min</span>
          </div>

          <div className="description">
            {film.overview}
          </div>
        </div>
      </div>
    </header>
  );
};

export default InfoPanel;