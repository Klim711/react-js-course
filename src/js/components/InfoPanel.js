import React, {Component} from 'react';
import {Link} from 'react-router-dom';

const InfoPanel = function(props) {
  return (
    <header className="row page-header">
      <div className="title-block">
        <h1 className="title">netflixroulette</h1>

        <Link to="/">
          <button className="btn">
            Search
          </button>
        </Link>
      </div>

      <div className="info-panel">
        <div className="info-panel__cover">
          <img src={props.film.coverUrl}/>
        </div>

        <div className="info-panel__text">
          <h2>{props.film.title} <span>{props.film.rating}</span></h2>

          <div className="additional-info">
            <span className="year">{props.film.year}</span>
            <span className="time">{props.film.time} min</span>
          </div>

          <div className="description">
            {props.film.description}
          </div>
        </div>
      </div>
    </header>
  );
};

export default InfoPanel;