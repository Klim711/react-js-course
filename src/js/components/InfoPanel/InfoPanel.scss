@import '../../../styles/constants.scss';

.title-block {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.info-panel {
  margin: 30px 0;
  display: flex;

  &__text {
    margin-left: 50px;

    h2 {
      color: $red;

      span {
        display: inline-block;
        color: $white;
        border: 1px solid $white;
        border-radius: 50%;
        width: 30px;
        height: 30px;
        line-height: 30px;
        text-align: center;
        margin-left: 20px;
      }
    }

    .additional-info {
      .time {
        margin-left: 30px;
      }
    }

    .description {
      margin-top: 30px;
    }
  }
}