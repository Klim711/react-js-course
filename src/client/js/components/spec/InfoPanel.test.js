import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import InfoPanel from '../InfoPanel';

describe('InfoPanel component', () => {
  it('matches to snapshot', () => {
    const film = {
      poster_path: 'mockPath',
      vote_average: 'mockVote',
      release_date: 'mockDate',
      runtime: 'mockRuntime',
      overview: 'mockOverview',
    };

    const renderedValue = renderer.create(
      <BrowserRouter>
        <InfoPanel film={film}/>
      </BrowserRouter>,
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
