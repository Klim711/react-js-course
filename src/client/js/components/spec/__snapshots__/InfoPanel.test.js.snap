// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel component matches to snapshot 1`] = `
<header
  className="row page-header"
>
  <div
    className="title-block"
  >
    <h1
      className="title"
    >
      netflixroulette
    </h1>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="btn"
      >
        Search
      </button>
    </a>
  </div>
  <div
    className="info-panel"
  >
    <div
      className="info-panel__cover"
    >
      <img
        src="mockPath"
      />
    </div>
    <div
      className="info-panel__text"
    >
      <h2>
         
        <span>
          mockVote
        </span>
      </h2>
      <div
        className="additional-info"
      >
        <span
          className="year"
        >
          mockDate
        </span>
        <span
          className="time"
        >
          mockRuntime
           min
        </span>
      </div>
      <div
        className="description"
      >
        mockOverview
      </div>
    </div>
  </div>
</header>
`;
